include/asterisk/res_pjsip_redirect.h

```diff
@@ -0,0 +1,146 @@
+/*
+ * Asterisk -- An open source telephony toolkit.
+ *
+ * Copyright (C) 2025, Commend International
+ *
+ * Maximilian Fridrich <m.fridrich@commend.com>
+ *
+ * See http://www.asterisk.org for more information about
+ * the Asterisk project. Please do not directly contact
+ * any of the maintainers of this project for assistance;
+ * the project provides a web site, mailing lists and IRC
+ * channels for your use.
+ *
+ * This program is free software, distributed under the terms of
+ * the GNU General Public License Version 2. See the LICENSE file
+ * at the top of the source tree.
+ */
+
+#ifndef _RES_PJSIP_REDIRECT_H
+#define _RES_PJSIP_REDIRECT_H
+
+#include <pjsip.h>
+
+/*!
+ * \brief Maximum number of redirect hops allowed
+ */
+#define AST_SIP_MAX_REDIRECT_HOPS 5
+
+/*!
+ * \brief Maximum size for a URI string
+ */
+#define AST_SIP_MAX_URI_SIZE 512
+
+/*!
+ * \brief Maximum number of redirect contacts to process
+ */
+#define AST_SIP_MAX_REDIRECT_CONTACTS 20
+
+/*!
+ * \brief Opaque structure for redirect context
+ *
+ * This structure encapsulates all state needed for handling
+ * SIP 3xx redirects, including visited URIs for loop detection,
+ * pending contacts for retry logic, and hop counting.
+ */
+struct ast_sip_redirect_context;
```

Comment:
Context has a rather specific meaning within Asterisk, and I must admit on reading this name I thought it meant dialplan context and was confused until the brief description sunk in. I think a different name would eliminate that confusion. "ast_sip_redirect_state"

Reply:
Good point, thanks and sorry about the confusion. I'll rename everything to state.